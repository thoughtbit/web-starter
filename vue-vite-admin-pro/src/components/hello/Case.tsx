import { defineComponent, ref, watchEffect } from "vue";
import type { PropType } from "vue";

const renderIcon = (item: any) => {
  return (
    <>
      <div>标题</div>
      <div>图片</div>
    </>
  );
};

export default defineComponent({
  name: "TestComponent",
  props: {
    value: String,
    // value: {
    //   type: String as PropType<string>,
    //   default: "",
    // },
    // list: {
    //   type: Array as PropType<string[]>,
    //   default: () => [],
    // },
    // isShow: {
    //   type: Boolean as PropType<boolean>,
    //   default: false,
    // },
  },
  emits: ["update:value"],
  setup(props, { emit }) {
    const local = ref("");

    watchEffect(() => {
      emit("update:value", local);
    });
    watchEffect(() => {
      local.value = props.value!;
    });

    return {
      local,
    };
  },
  render() {
    return (
      <a-select v-model={[this.local, 'value']}>
        <a-select-option value="aaa">aaa</a-select-option>
      </a-select>
    );
  },
});
