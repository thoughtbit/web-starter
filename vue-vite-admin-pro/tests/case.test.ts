import { shallowMount } from "@vue/test-utils";
import { expect, test } from "vitest";
import Case from "../src/components/hello/Case";

test("mount component", () => {
  const wrapper = shallowMount(Case, {
    props: {
      value: "test",
    },
    global: {
      stubs: ["select", "option"],
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
