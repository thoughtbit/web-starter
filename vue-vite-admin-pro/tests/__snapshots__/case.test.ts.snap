// Vitest Snapshot v1

exports[`mount component 1`] = `
"<a-select value=\\"test\\">
  <a-select-option value=\\"aaa\\">aaa</a-select-option>
</a-select>"
`;
